api/outdated/outdated/models.py

```diff
@@ -10,57 +14,141 @@
     "up_to_date": "UP-TO-DATE",
     "undefined": "UNDEFINED",
 }
-STATUS_CHOICES = [(status, status) for _, status in STATUS_OPTIONS.items()]
+
+STATUS_CHOICES = [(_, _) for _ in STATUS_OPTIONS.values()]
+
+PROVIDER_OPTIONS = {
+    "PIP": "https://pypi.org/pypi/%s/json",
+    "NPM": "https://registry.npmjs.org/%s",
+}
+
+PROVIDER_CHOICES = [(provider, provider) for provider in PROVIDER_OPTIONS.keys()]
+
+
+def get_yesterday():
+    return timezone.now() - timedelta(days=1)
+
+
+def get_latest_patch_version(release_version):
+    url = (
+        PROVIDER_OPTIONS[release_version.dependency.provider]
+        % release_version.dependency.name
+    )
+    json = get(url).json()
+    try:
+        return SemVer.parse(
+            sorted(
+                [
+                    version
+                    for version in (json.get("releases") or json.get("versions")).keys()
+                    if version.startswith(release_version.version + ".")
+                ],
+            )[-1]
+        ).patch
+    except IndexError:
+        return 0
 
 
 class Dependency(UUIDModel):
-    name = models.CharField(max_length=100, unique=True)
+    name = models.CharField(max_length=100)
+    provider = models.CharField(max_length=10, choices=PROVIDER_CHOICES)
 
     class Meta:
         ordering = ["name", "id"]
+        unique_together = ("name", "provider")
 
     def __str__(self):
         return self.name
 
 
-class DependencyVersion(UUIDModel):
-
+class ReleaseVersion(UUIDModel):
     dependency = models.ForeignKey(Dependency, on_delete=models.CASCADE)
-    version = models.CharField(max_length=100)
-    release_date = models.DateField()
-    end_of_life_date = models.DateField()
+    major_version = models.IntegerField()
+    minor_version = models.IntegerField()
+    _latest_patch_version = models.IntegerField(editable=False, null=True, blank=True)
+    last_checked = models.DateTimeField(
+        editable=False,
+        null=True,
+        blank=True,
+        default=get_yesterday,
+    )
+
     status = models.CharField(max_length=10, choices=STATUS_CHOICES, editable=False)
```

Comment:
This might be obsolete as there is a property that has the same name.

Reply:
Done in https://github.com/adfinis/Outdated/pull/21/commits/72c3c3f8c0e0b29c7fa2be5bb2544314bfc78831